app/src/main/java/com/sun/quickquotes/base/BaseViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.sun.quickquotes.base
+
+import androidx.annotation.StringRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sun.quickquotes.R
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.launch
+
+abstract class BaseViewModel : ViewModel() {
+
+    private val _isLoading = MutableLiveData<Boolean>().apply { value = false }
```

Comment:
nếu vội thì thôi, nh rảnh thì xem qua bài này nhé https://medium.com/androiddevelopers/livedata-with-snackbar-navigation-and-other-events-the-singleliveevent-case-ac2622673150
với case này nên dùng event nha